metricflow/dataflow/builder/dataflow_plan_builder.py

```diff
@@ -817,78 +812,78 @@ def _find_dataflow_recipe(
         time_range_constraint: Optional[TimeRangeConstraint] = None,
     ) -> Optional[DataflowRecipe]:
         linkable_specs = linkable_spec_set.as_tuple
-        potential_source_nodes: Sequence[BaseOutput]
-        input_time_spine_source_node: Optional[MetricTimeDimensionTransformNode] = None
+        candidate_nodes_for_left_side_of_join: Sequence[BaseOutput]
+        candidate_nodes_for_right_side_of_join: Sequence[BaseOutput]
         if measure_spec_properties:
-            source_nodes = self._source_nodes
-            potential_source_nodes = self._select_source_nodes_with_measures(
-                measure_specs=set(measure_spec_properties.measure_specs), source_nodes=source_nodes
+            candidate_nodes_for_right_side_of_join = self._source_node_set.source_nodes_for_metric_queries
+            candidate_nodes_for_left_side_of_join = self._select_source_nodes_with_measures(
+                measure_specs=set(measure_spec_properties.measure_specs),
+                source_nodes=self._source_node_set.source_nodes_for_metric_queries,
             )
             default_join_type = SqlJoinType.LEFT_OUTER
         else:
-            # Only read nodes can be source nodes for queries without measures
-            source_nodes = list(self._read_nodes)
-            potential_source_nodes = self._select_read_nodes_with_linkable_specs(
-                linkable_specs=linkable_spec_set, read_nodes=self._read_nodes
-            )
-            # `metric_time` does not exist if there is no metric in the query.
-            # In that case, we'll use the time spine table to represent `metric_time` values.
-            requested_metric_time_specs = [
-                time_dimension_spec
-                for time_dimension_spec in linkable_spec_set.time_dimension_specs
-                if time_dimension_spec.element_name == self._metric_time_dimension_reference.element_name
-            ]
-            if requested_metric_time_specs:
-                input_time_spine_source_node = self._time_spine_source_node
-                # Add time_spine source node to potential source nodes
-                potential_source_nodes = list(potential_source_nodes) + [self._time_spine_source_node]
+            # For the case where we want to find a recipe for a group-by-item and metric_time, we add the time spine
+            # node (which has metric_time for all date values), and set the default join type between nodes to
+            # FULL_OUTER. This will generate a recipe where you join the node containing the group-by-item and the
+            # time spine node to get the group-by-item with metric_time.
+            candidate_nodes_for_right_side_of_join = list(self._source_node_set.source_nodes_for_group_by_item_queries)
```

Comment:
This is actually a behavior change, though it might not result in a change in output. We previously only added time spine to the source nodes if `metric_time` was queried without metrics. Now time spine will be included in the source nodes any time only group by items are queried.
I think that should be fine, but just want to double check that the time spine wouldn't be used unless `metric_time` is queried. Would be something to double check in node evaluation.

Reply:
It's fine to always have it there as if `metric_time` is not queried, then the `NodeEvaluatorForLinkableInstances` won't select it as a node to join with.

Also, there were no changes to the SQL snapshots as a result.